src/main/java/com/carmen/app/services/CarService.java

```diff
@@ -1,43 +1,59 @@
 package com.carmen.app.services;
 
+import java.util.List;
 import java.util.UUID;
 
+import javax.ejb.Stateless;
 import javax.persistence.EntityManager;
 import javax.persistence.EntityManagerFactory;
 import javax.persistence.Persistence;
-
+import javax.persistence.TypedQuery;
 
 import com.carmen.app.entities.Car;
+import com.carmen.app.exceptions.CarNotFoundException;
 
+@Stateless
 public class CarService {
 
 	EntityManagerFactory emf = Persistence.createEntityManagerFactory("carP");
 	private EntityManager em = emf.createEntityManager(); 
 	
-	private Car car = new Car();
 	
-	public Car getCar(UUID id) {
-	 em.find(Car.class, id);
-		if(car != null) {
-			return car;
+	public List<Car> getCars(){
+		TypedQuery <Car>  query = this.em.createNamedQuery("Car.FindCars", Car.class);
+		return query.getResultList();
+		
+	}
+	
+	public Car getCar(UUID id) throws CarNotFoundException{
+		Car car =  this.em.find(Car.class, id);
+		if(car == null) {
+			throw new CarNotFoundException("Car with id " + id + " not found");
 		}
-	return null;
+		return car;
+		
 	}
 	
 	
 	public void createCar(Car car) {
-		em.persist(car);
+		this.em.persist(car);
 	
 	}
 	
-	public void updateCar(UUID id) {
-	em.find(Car.class, id);
-		em.merge(car);
+	public void updateCar(Car car) throws CarNotFoundException {
+		Car car_update = getCar(car.getId());
+		this.em.merge(car_update);
+
 	}
 	
-	public void deleteCar(UUID id) {
-		em.find(Car.class, id);
-		em.remove(car);
+	public void deleteCar(UUID id) throws CarNotFoundException{
+		Car car = this.getCar(id);
+		if(car == null) {
```

Comment:
In the getCar you're already checking if the car was found and throwing an exception.